File: ruby/lib/Sikaru/tool_providers/types/attach_source_tool_skill_request.rb
# frozen_string_literal: true

module Sikaru
  module ToolProviders
    module Types
      class AttachSourceToolSkillRequest < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :tool_provider_id, -> { String }, optional: false, nullable: false

        field :capability_refs, -> { Internal::Types::Array[String] }, optional: false, nullable: false

        field :description, -> { String }, optional: false, nullable: false

        field :source, -> { Sikaru::Types::ToolSkillSourceRequest }, optional: false, nullable: false
      end
    end
  end
end

File: typescript/src/api/resources/toolProviders/client/requests/AttachSourceToolSkillRequest.ts

import * as Sikaru from "../../../../index.js";

/**
 * @example
 *     {
 *         capability_refs: ["capability_refs"],
 *         description: "description",
 *         source: {
 *             kind: "kind"
 *         }
 *     }
 */
export interface AttachSourceToolSkillRequest {
    capability_refs: string[];
    description: string;
    source: Sikaru.ToolSkillSourceRequest;
}
